### ISSUE TITLE
🐜 Accessibility Bug: Error handling in Forms

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/gc-digital-talent#5416]
## ❌ Current behaviour (issue)
Error messages not helpful. Sometimes the form fields do present the errors, but do not give instructions on how to fix the errors. Error Block only contains unsaved changes. 

## Relevant WCAG Criteria: 

Understanding SC 3.3.3: Error Suggestion (Level AA): If an input error is automatically detected and suggestions for correction **are known**, then the suggestions are provided to the user, unless it would jeopardize the security or purpose of the content.

[G177: Providing suggested correction text](https://www.w3.org/WAI/WCAG22/Techniques/general/G177)

### How to reproduce
1. Navigate to Skills and Experience section of the application profile section using any screen reader with keyboard navigation.
2. Leave the organization field empty and enter a bogus, non conformant date. 
3. Navigate to the save and go back button and activate it. 
4. understand how to fix your errors by only listening to the screen reader's summary of the error blocks.
5. Activate the error links to modify the required fields. 
### Screen Captures
![#5416 Screen capture.JPG](https://images.zenhubusercontent.com/625076b8f67c446dd188ffe5/e354dd3f-e7cd-4daf-8624-e0d58d17fd14)

![#5416 Screen Capture 2.JPG](https://images.zenhubusercontent.com/625076b8f67c446dd188ffe5/b3c6a655-4021-47e1-a2b1-78f291827fa6)

## ✔ Expected behaviour (fix)

1. Error block messages provide key information to screen reader users about how to fix the errors with any known corrections.

### Acceptance Criteria
- [ ] Epic is created to capture this work
- [ ] At least one actionable issue is added to the epic

**Comment by petertgiles** [comment]
Hi Rosalie, thanks for this one.  Could I ask for some clarifications?

For the second point, the issue is that while it scrolls to the error-ed field the focus is not actually in the input, right?

For the first point, you mention "provide key information to screen reader users".  What is this exactly?  Using the _Organization_ field as an example, what information should be provided in what way?

**Comment by RM-1978** [comment]
Hi Peter, 
Thanks for your note! 

- You are right about the second item. Confirming that the focus does not actually land in the input according to Fable and my reproduction.  
- For the other item @substrae, what are your thoughts on a standard error message " Please enter [type of information required] could help the user enter the key information we are looking for in that field?

  Using the _Organization_ field as an example, we could direct the user with the following error statement: "Please enter the name of your employer." 

- Here are the other directions that I suggest for the other required fields: 
    - [ ] For My role, I would enter: "Please enter the name of the role you held."
    - [ ] Start Date: "Please enter the date you started in this role. Enter the date manually by month, day year, or use the spin box to select a date"
    - [ ] End Date: "Please enter the last date you worked in this role or position. Enter the month, day year or use the spin box 
    to select a date."

**Comment by petertgiles** [comment]
Thanks Rosalie.  The second item seems clear and actionable right away.  I'm going to split that off into a new issue while a solution is designed for the help messages.

**Comment by mnigh** [comment]
Steps to move this forward:
1. create an epic out of this issue since scope is huge
2. each form has its own issue within epic
3. new forms should follow this pattern (@GCTC-NTGC/designers, @substrae)

**Comment by substrae** [comment]
Just gonna pop some context in here:
- I'm going to go ahead and account for these states on default input types in Figma ([found here](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=5-13&mode=design))
- Once the designs are ready, I'll reply again here, and we can apply them universally
- We can then identify instances where inputs have custom validation that extends outside the usual default states and collaborate on messaging for each instance

**Comment by substrae** [comment]
After thinking about this a little more, I'm not sure there's much I can do in terms of Figma designs. This seems to be more of a content issue and will require a pretty significant copy lift and translation if we intend to provide custom error messages for every single validation state for every single input on the site. Ideally @gobyrne can chime in terms of a path forward here before we move on anything.

**Comment by RM-1978** [comment]
@gobyrne As we said, happy to buckle in and review the suggestions we might want to design to help the user understand how to fix the errors they might make in the fields. 💯

**Comment by RM-1978** [comment]
@mnigh Let's create an epic for all the error messages. Then, we can find error suggestions for each one.

**Comment by mnigh** [comment]
> @mnigh Let's create an epic for all the error messages. Then, we can find error suggestions for each one.

@RM-1978 This issue is already an epic.

**Comment by RM-1978** [comment]
> > @mnigh Let's create an epic for all the error messages. Then, we can find error suggestions for each one.
> 
> @RM-1978 This issue is already an epic.
 Perfect.

**Comment by mnigh** [comment]
@gobyrne is there a reason this issue is [pinned in github](https://docs.github.com/en/issues/tracking-your-work-with-issues/pinning-an-issue-to-your-repository)?

**Comment by RM-1978** [comment]
Closed in favour of  #8851.  a design ticket for instructional patterns for error messages outside the **required field** error will be created.
